server/gamenode/gameserver.js

```diff
@@ -213,6 +213,9 @@ class GameServer {
 
     onStartGame(pendingGame) {
         let game = new Game(pendingGame, { router: this, titleCardData: this.titleCardData, cardData: this.cardData, packData: this.packData, restrictedListData: this.restrictedListData });
+        game.on('onTimeExpired', () => {
+            this.sendGameState(game);
```

Comment:
Having the Game Server listen to events on game makes me slightly nervous. Though I don't have a good alternative (e.g. you'd have to pass the `sendGameState` method into the game object itself)

Reply:
I think I get your concern but it kind of fits the pattern. The server is what sends status updates to the connected users, I’d be less happy about the game itself starting to manage state updates

Reply:
Yeah, sending game state should be handled here. The alternative is to send in a `handleTimeExpired` method to `Game` that happens to be implemented as sending state. But that's not a lot different than this.